code/game/machinery/vending.dm

```diff
@@ -443,16 +467,21 @@
 			return
 
 	if(href_list["remove_coin"])
-		if(!coin)
-			to_chat(usr, "<span class='notice'>There is no coin in this machine.</span>")
+		if(!(coin || bill))
+			to_chat(usr, "<span class='notice'>There is no money in this machine.</span>")
 			return
-
-		coin.loc = loc
-		if(!usr.get_active_held_item())
-			usr.put_in_hands(coin)
-		to_chat(usr, "<span class='notice'>You remove [coin] from [src].</span>")
-		coin = null
-
+		if(coin)
+			coin.loc = loc
+			if(!usr.get_active_held_item())
+				usr.put_in_hands(coin)
+				to_chat(usr, "<span class='notice'>You remove [coin] from [src].</span>")
+				coin = null

```

Comment:
else
    coin.forceMove(get_turf(src))

unindent the to_chat and coin = null lines